### PR TITLE
Build a way to visualise completed projects on the Project index page

### PR DESCRIPTION
**Pull request by lozette**
## Changes

We want to make completed projects more easily findable on the index page.

Previously, completed (or "closed") projects were pushed to the end of the
project list, so it wasn't entirely obvious where they were. This commit splits
the index page into two actions, and presents them on separate tabs on the
project index page.

The first tab is "In progress" projects, i.e projects which are open (do not
have a `completed_at` date). The second tab is for "Completed" projects. Both
tabs have their own route & action.

As a side effect of this, the layout for the index page has been put into a
partial, to be included in the views for both the `index` and `completed`
actions.

The relevant tests have been updated, and a new test for the completed project
view has been added.

Screenshot 2022-11-15 at 11 45 52

Trello: https://trello.com/c/liZ2Yqcv/795-build-a-way-to-visualise-closed-project 

## Checklist

- [x] Attach this pull request to the appropriate card in Trello.
- [x] Update the `CHANGELOG.md` if needed.

### CODE REVIEW

**Comment by lozette** [comment]
[Build a way to visualise closed project](https://trello.com/c/liZ2Yqcv/795-build-a-way-to-visualise-closed-project)

File: config/routes.rb [hunk 1]
```diff
@@ -22,6 +22,8 @@
     get "/delete", action: :confirm_destroy
   end
 
+  get "completed", to: "projects#completed"
```
Comment (lozette) [inline config/routes.rb:25]: I'm not 100% convinced this is the best route. Would `projects/completed` make more sense?

File: spec/models/project_spec.rb [hunk 2]
```diff
@@ -301,4 +301,34 @@
       end
     end
   end
+
+  describe "completed scope" do
```
Comment (jkempster34) [inline spec/models/project_spec.rb:305]: Might be nice to wrap these in a `describe "Scopes"` which is what we have done in the other spec files

File: config/routes.rb (hunk 1 above)
Comment (jkempster34) [reply config/routes.rb:25]: I think it would make more sense!

**Review by jkempster34 — CHANGES_REQUESTED** [review]
Looks great, like the tabs! :pinched_fingers:

Found a few things:
- One of the commit titles is a bit too long for GitHub and is wrapping
- I defo prefer `projects/completed` to `completed`
- There is some code that we can remove now. The `by_completed_state` scope, and where it's called in the `project_policy`

File: app/models/project.rb [hunk 3]
```diff
@@ -33,6 +33,9 @@ class Project < ApplicationRecord
   # and see if you can use Arel to build a proper query.
   scope :by_completed_state, -> { order(:completed_at) }
 
+  scope :completed, -> { where.not(completed_at: nil) }
+  scope :open, -> { where(completed_at: nil) }
```
[the remark this answers is not in the document]
Comment (lozette) [reply app/models/project.rb:37]: Oh not seen that before but not a fan, I think it makes it more obscure!

**Comment by jkempster34** [comment]
> Added text to display when the tabs are empty
> 
> Screenshot 2022-11-15 at 15 47 22

Nice! Is it worth matching it to the `No notes` messages and putting it in an inset text?

image

**Comment by lozette** [comment]
> 

I've added it as per https://lucid.app/lucidspark/59a224c0-3bb3-4a0f-ace4-199138d009f5/edit?invitationId=inv_a2eb4a53-99fa-4945-93a6-aeb92467c134# so I'm not sure - @Rich-Cooley any thoughts?

File: app/models/project.rb (hunk 3 above)
[the remark this answers is not in the document]
Comment (jkempster34) [reply app/models/project.rb:31]: Agree!

File: app/views/projects/index/_project_index.html.erb [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    <% if policy(:project).new? %>
+      <p><%= link_to t("project.index.new_button.text"),
+               new_project_path,
+               class: "govuk-button",
+               data: {module: "govuk-button"} %></p>
+    <% end %>
+
+    <%= render partial: "shared/project_index_sub_navigation" %>
+
+    <%- if @projects.empty? %>
```
Comment (jkempster34) [inline app/views/projects/index/_project_index.html.erb:19]: We haven't used these `-` anywhere else, I'm not sure what they do, or do they just do the same as omitting the `-`?

File: config/routes.rb [hunk 5]
```diff
@@ -22,6 +22,8 @@
     get "/delete", action: :confirm_destroy
   end
 
+  get "projects/completed", to: "projects#completed"
```
Comment (jkempster34) [inline config/routes.rb:25]: I think we can put this inside the block below like:

```ruby
resources :projects, only: %i[index show new create] do
  collection do
    get "completed"
  end
```

**Review by jkempster34 — APPROVED** [review]
Left two picky stylistic comments, and one about the routing (I looked into this recently to tidy it up).

Looks great 💯

File: app/views/projects/index/_project_index.html.erb (hunk 4 above)
Comment (lozette) [reply app/views/projects/index/_project_index.html.erb:19]: You know what I think these are quite old erb style and I haven't used Ruby for a year so I'm out of date 😆

**Pull request merged**